content/doc/book/using/best-practices.adoc

```diff
@@ -0,0 +1,143 @@
+
+.GitLab group folders
+video::it6TOeQ6EHg[youtube,width=800,height=420]
+
+.Bitbucket project folders
+video::85b6fiVolfk[youtube,width=800,height=420]
+
+.Gitea organization folders
+video::NO3sZWRxgQM[youtube,width=800,height=420]
+
+== Use multibranch Pipelines
+
+If something prevents you from using organization folders, then use multibranch Pipelines.
+
+Insert more wisdom here ...
+
+See the link:/doc/book/pipeline-as-code/[multibranch Pipelines] documentation for more details.
+
+== Use Pipeline
+
+Insert Pipeline wisdom here
+
+See the link:/doc/book/pipeline/[Pipeline] documentation for more details.
+
+video::IOUm1lw7F58[youtube,width=800,height=420]
+
+== Use reports effectively
+
+Charts and graphs show trends and patterns that can highlight project status and progress.
+Automated test results for unit tests, integration tests, and end to end tests can highlight brittleness or instability.
+Coverage reports can illustrate areas that automated tests are not executing.
+Compiler warning messages can be a first indication of a problem.
+Static analysis tools can report risky code or code that may have security risks.
+Performance test results can idenitfy delays or problem areas.
+
+The plugin:warnings-ng[warnings next generation] plugin provides convenient access to many reports, including:
+
+* compiler warnnings and errors (like gcc, clang, javac, golang, etc.)
+* static analysis warnings and errors (like spotbugs, checkstyle, pmd, lint, cpd, Simian, etc.)
+* code coverage reports
+
+video::tj3xYFA6Q2o[youtube,width=800,height=420]
+
+== Show failures to the right people
+
+Tune the notifications from failing and unstable jobs so that the notifications reach the right people without distracting others.
+Many Jenkins users prefer to only be notified when a failure is likely their fault.
+They correctly assume that if they did not cause the failure, they are not the best person to investigate the failure.
+
+Refine your notification system so that new test failures first notify the most recent committers, since they are the most likely cause of the issue.
+
+.Sending Slack notifications
+video::EDVZli8GdUM[youtube,width=800,height=420]
+
+== Build on agents
+
+Use agents to perform builds instead of running builds on the controller.
+It is much safer to use agents and much more scalable.
+
+More details are available in the link:/doc/book/security/controller-isolation/[controller isolation] documentation.
+
+== Back up regularly
+
+Even the most reliable of systems will have failures.
+Prepare and regularly check the health of your backups.
+
+More details in the link:/doc/book/system-administration/backing-up/[backup documentation].
+
+== Use simple project names
+
+Jenkins uses project names for folders related to the project.
+Many tools cannot handle spaces, dollar signs, or similar characters in file paths.
+Limit yourself to alphanumeric characters [a-zA-Z0-9_-]+ in project names.
```

Comment:
```suggestion
Limit yourself to alphanumeric characters [a - z, A - Z, 0 - 9, _, -, +] in project names.
```
suggestion to reformat characters for clarity

Reply:
The string is formatted as though it were a regular expression.  I'd rather remove the regex syntax than insert the white space, since that makes it no longer a regex.

Reply:
That makes sense to me @MarkEWaite! My original intent was to ensure that the `+` was contained within the brackets, so I should have just left it at that.